App.js

```diff
@@ -1,9 +1,363 @@
+    setGameState(() => {
+      return {
+        roll: null,
+        even: false,
+        odd: false,
+        position: 0,
+        message: 'Roll again',
+        optMessage: 'Kim',
+        score: 0,
+        board: JSON.parse(JSON.stringify(newBoard)),
+      };
+    });
+  }, []);
+
+  // press Even button
+  const pressEven = useCallback(() => {
+    setGameState(prevGameState => {
+      let workEven;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if (prevGameState.even === true) {
+        workEven = false;
+      } else {
+        workEven = true;
+      }
+      if (workEven === true && prevGameState.odd === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'cannot select both ODD and EVEN';
+        workEven = false;
+      }
+      if (workEven === false && prevGameState.odd === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'ODD selected';
+      }
+      if (prevGameState.odd === false && workEven === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'EVEN selected';
+      }
+      return {
+        roll: prevGameState.roll,
+        even: workEven,
+        odd: prevGameState.odd,
+        position: prevGameState.position,
+        message: workMessage,
+        optMessage: workOptMessage,
+        score: prevGameState.score,
+        board: prevGameState.board,
+      };
+    });
+  }, []);
+
+  // press Odd button
+  const pressOdd = useCallback(() => {
+    setGameState(prevGameState => {
+      let workOdd;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if (prevGameState.odd === true) {
+        workOdd = false;
+      } else {
+        workOdd = true;
+      }
+      if (workOdd === true && prevGameState.even === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'cannot select both ODD and EVEN';
+        workOdd = false;
+      }
+      if (workOdd === true && prevGameState.even === false) {
+        workMessage = 'Roll again';
+        workOptMessage = 'ODD selected';
+      }
+      if (workOdd === false && prevGameState.even === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'EVEN selected';
+      }
+      return {
+        roll: prevGameState.roll,
+        even: prevGameState.even,
+        odd: workOdd,
+        position: prevGameState.position,
+        message: workMessage,
+        optMessage: workOptMessage,
+        score: prevGameState.score,
+        board: prevGameState.board,
+      };
+    });
+  }, []);
+
+  // press Roll button
+  const pressRoll = useCallback(() => {
+    setGameState(prevGameState => {
+      let workBoard = prevGameState.board.slice();
+      let randomNumber = Math.floor(Math.random() * 6) + 1;
+      let scoreAdj = 1;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if ((randomNumber === 1 || randomNumber === 3 || randomNumber === 5) && prevGameState.odd === true) {
```

Comment:
To figure out if a number is even or odd you can use modulus (% symbol is used for this in programming).
It's a math operator that returns the reminder in a division operation.
So 16 % 3 = 1, since 16 /3 = 5 reminder 1.

To figure out parity (even or odd) you can do yourNumber % 2, if it's 0 then it's even, if it's 1 then it's odd.

Reply:
these should probably all be `else if` not that any of these if's should be hit twice, but more readable when you make it explicit

Reply:
DONE.